spec.html

```diff
@@ -36247,16 +36247,24 @@ <h1>
           <h1>
             GetStringIndex (
               _S_: a String,
-              _e_: a non-negative integer,
+              _cpIndex_: a non-negative integer,
             ): a non-negative integer
           </h1>
           <dl class="header">
+            <dt>description</dt>
+            <dd>It interprets _S_ as a sequence of UTF-16 encoded code points, as described in <emu-xref href="#sec-ecmascript-language-types-string-type"></emu-xref>, and returns the code unit index corresponding with code point index _cpIndex_, clipped to a maximum position just beyond the end of _S_.</dd>
           </dl>
           <emu-alg>
             1. If _S_ is the empty String, return 0.
-            1. Let _codepoints_ be StringToCodePoints(_S_).
-            1. Let _eUTF_ be the smallest index into _S_ that corresponds to the character at element _e_ of _codepoints_. If _e_ is greater than or equal to the number of elements in _codepoints_, then _eUTF_ is the number of code units in _S_.
-            1. Return _eUTF_.
+            1. Let _len_ be the length of _S_.
+            1. Let _cuCount_ be 0.
+            1. Let _cpCount_ be 0.
+            1. Repeat, while _cuCount_ &lt; _len_,
+              1. If _cpCount_ = _cpIndex_, return _cuCount_.
+              1. Let _cp_ be CodePointAt(_S_, _cuCount_).
+              1. Set _cpCount_ to _cpCount_ + 1.
+              1. Set _cuCount_ to _cuCount_ + _cp_.[[CodeUnitCount]].
```

Comment:
Nit: I'd switch this step and the previous, so that `cp` is used immediately after it's defined. Plus then it's consistent with the definitions, which occur in the order "cuCount" then "cpCount".